### PR TITLE
Added Codelabs schema [User, Maps, Node & Sections]

### PR DESCRIPTION
**Pull request by chandratop**

### CODE REVIEW

File: server/models/Codelabs/Map.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+const MapSchema = new mongoose.Schema({
+  nodes: [{
+    type: Schema.Types.ObjectId,
+    ref: 'Node'
+  }],
+  author: {
+    type: String,
+    required: true,
+  },
+  time: {
```
Comment (parthvshah) [inline server/models/Codelabs/Map.js]: Let's change this to createdTime. We may need some more functionality in the future. Also add a default value for this field. It's good practice for all time related fields to have a value defaulting to now.

File: server/models/Codelabs/Map.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+const MapSchema = new mongoose.Schema({
+  nodes: [{
+    type: Schema.Types.ObjectId,
+    ref: 'Node'
+  }],
+  author: {
```
Comment (parthvshah) [inline server/models/Codelabs/Map.js]: Author is a reference to the user.

File: server/models/Codelabs/Map.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    required: true,
+  },
+  time: {
+    type: Date,
+  },
+  tags: [{
+    type: String,
+    required: true
+  }],
+  groups: {
```
Comment (parthvshah) [inline server/models/Codelabs/Map.js]: Change to an array and make it a reference to groups.

File: server/models/Codelabs/Node.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+const NodeSchema = new mongoose.Schema({
+  sections: [{
+    type: Schema.Types.ObjectId,
+    ref: 'Sections'
+  }],
+  topics: {
```
Comment (parthvshah) [inline server/models/Codelabs/Node.js]: Change to an array.

File: server/models/Codelabs/Sections.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require('mongoose');
+var Schema = mongoose.Schema;
+
+const SectionsSchema = new mongoose.Schema({ // TODO
+  type: {
+    type: Number,
+    required: true,
+  },
+  author: {
```
Comment (parthvshah) [inline server/models/Codelabs/Sections.js]: Change to a reference to the user.

File: server/models/Codelabs/Sections.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    type: String,
+    required: true,
+  },
+  time: {
```
Comment (parthvshah) [inline server/models/Codelabs/Sections.js]: Rename to createdTime and set default value.

File: server/models/Codelabs/Sections.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    type: Date,
+  },
+  tags: [{
+    type: String,
+    required: true
+  }],
+  groups: {
```
Comment (parthvshah) [inline server/models/Codelabs/Sections.js]: Change to a reference to groups.

File: server/models/Codelabs/Sections.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    type: String,
+    required: true,
+  },
+  levels: {
```
Comment (parthvshah) [inline server/models/Codelabs/Sections.js]: Rename to 'level'.

File: server/models/Codelabs/Sections.js (hunk 7 above)
Comment (parthvshah) [reply server/models/Codelabs/Sections.js]: Still need to make this an array. Multiple groups are possible.

Comment (chandratop) [reply server/models/Codelabs/Sections.js]: Made groups as an array of references to Group.

**Pull request merged**